classes/bridges/across.js

```diff
@@ -19,16 +24,52 @@ class AcrossBridge extends BaseBridge {
     "op mainnet": "0x6f26Bf09B1C792e3228e5467807a900A503c0281",
     polygon: "0x9295ee1d8C5b022Be115A2AD3c30C72E34e7F096",
   };
+
   constructor() {
     super("across");
     this.sdk = createAcrossClient({
       integratorId: "0xaaff", // 2-byte hex string
       chains: [optimism, arbitrum, polygon, base],
     });
+    this.isInitialized = false;
+    this.feeCosts = null;
   }
+
   async init() {
     return;
   }
+
+  async fetchFeeCosts(
+    account,
+    fromChainId,
+    toChainId,
+    inputToken,
+    targetToken,
+    inputAmount,
+    tokenPrices,
+  ) {
+    const route = {
+      originChainId: fromChainId,
+      destinationChainId: toChainId,
+      inputToken: inputToken,
+      outputToken: targetToken,
+    };
+    const quote = await this.sdk.getQuote({
+      route,
+      inputAmount: inputAmount,
+    });
+    const tokenDecimal = await getTokenDecimal(
+      quote.deposit.inputToken,
+      CHAIN_ID_TO_CHAIN[fromChainId].name,
+    );
+    const feeInToken = ethers.utils.formatUnits(
+      quote.fees.totalRelayFee.total,
+      tokenDecimal,
+    );
+    this.feeCosts = feeInToken * tokenPrices;
+    return this.feeCosts;
```

Comment:
👍

Reply:
@fantzulin actually you need to add the transaction costs to `feeCosts`. The total fee costs consists of `bridge fee + transaction costs (交易磨損)`
https://discord.com/channels/1108070617720299540/1108070622577315893/1356496649471660103

Reply:
since we've request API at `const quote = await this.sdk.getQuote`, we can reuse the result to calcualte:
1. bridge fee (done)
2. transaction costs (not implemented yet)